model/metrics/http.yaml

```
@@ -124,3 +124,147 @@ groups:
       is often, but not always, present as the [Content-Length](https://www.rfc-editor.org/rfc/rfc9110.html#field.content-length)
       header. For requests using transport encoding, this should be the compressed size.
     extends: metric_attributes.http.client
+
+  - id: metric.http.client.open_connections
+    type: metric
+    metric_name: http.client.open_connections
+    stability: experimental
+    brief: "Number of outbound HTTP connections that are currently active or idle on the client."
+    instrument: updowncounter
+    unit: "{connection}"
+    attributes:
+      - ref: http.connection.state
+        requirement_level: required
+      - ref: network.peer.address
+        requirement_level: recommended
+        brief: Remote IP address of the socket connection.
+        examples: ["10.1.2.80"]
+      - ref: network.protocol.version
+        requirement_level: recommended
+        brief: HTTP protocol version of the connection in the connection pool.
+        note: >
+          HTTP 1.0 and 1.1 requests share connections in the connection pool and are both reported as version `1.1`.
+          So, the `network.protocol.version` value reported on connection metrics is different than the one reported
+          on request-level metrics or spans for HTTP 1.0 requests.
+        examples: ["1.1", "2", "3"]
+      - ref: server.address
+        requirement_level: required
+        brief: >
+          Host identifier of the ["URI origin"](https://www.rfc-editor.org/rfc/rfc9110.html#name-uri-origin) HTTP request is sent to.
+        note: >
+          If an HTTP client request is explicitly made to an IP address, e.g. `http://x.x.x.x:8080`, then
+          `server.address` SHOULD be the IP address `x.x.x.x`. A DNS lookup SHOULD NOT be used.
+      - ref: server.port
+        requirement_level:
+          conditionally_required: If not the default (`80` for `http` scheme, `443` for `https`).
+        brief: >
+          Port identifier of the ["URI origin"](https://www.rfc-editor.org/rfc/rfc9110.html#name-uri-origin) HTTP request is sent to.
```

We're requiring port on stable OTel http metrics now (was a last minute change), it's tracked here on the .NET side  in https://github.com/dotnet/runtime/issues/94829

```suggestion
        requirement_level: required
```

---

Ah thanks for the pointer. 

Should we then just inherit from `[metric_attributes.http.client`? ](https://github.com/open-telemetry/semantic-conventions/blob/main/model/metrics/http.yaml#L21)

That includes all of these +

- http.request.method
- http.response.status_code
- error.type
- network.protocol.name
- network.protocol.version

---

I think we can't extend common attributes on this metric because it does not include `http.response.status_code` or `error.type` (open_connections counter couldn't and shouldn't)

---

So I think what you have is the best we can do today (we don't inherit req levels and briefs from http because we don't extend anything and therefore end up inheriting generic req level and brief :(